api/all/src/main/java/io/opentelemetry/api/baggage/ImmutableEntryMetadata.java

```diff
@@ -24,6 +26,7 @@
    */
   static ImmutableEntryMetadata create(String metadata) {
     if (metadata == null) {
+      API_USAGE_LOGGER.finest("ImmutableEntryMetadata.create: metadata is null");
```

Comment:
I think instead of exposing the `Logger` we can expose a `log` method that calls finest which we are calling everywhere.

Reply:
Also instead of logging just a string with some location information, let's log a throwable to provide a complete stack trace 

`log(Level.FINEST, "metadata is null", new NullPointerException())`